core/src/nl/tudelft/watchdog/core/logic/ui/listeners/staticanalysis/CheckStyleChecksMessagesFetcher.java

```diff
@@ -12,52 +12,39 @@
 import nl.tudelft.watchdog.core.util.WatchDogUtilsBase;
 
 /**
- * Hello and welcome to this class full of reflection, hacks and other shenanigans.
- * Before you ask: yes this is necessary. To explain why, please sit back, relax (while you still can)
- * and get to know the absolute horror of having multiple classloaders.
+ * Warning: This class is full of classloader fixes, reflection, hacks to support Checkstyle warnings in IntelliJ
+ * and Eclipse. There are Checkstyle IDE plugins (for Eclipse and IntelliJ) and the plain Checkstyle jar itself.
+ * The main complication comes from IntelliJ depending on Checkstyle via dynamically loading its jar over
+ * a {@link ClassLoader}. In Eclipse, we do not know whether the user has installed the Checkstyle plugin at all,
+ * which means we have to dynamically load the Checkstyle classes, too.
```

Comment:
how about changing to "we can also only dynamically depend on the Checkstyle classes."?